src/envoy/auth/controller.h

```diff
@@ -0,0 +1,78 @@
+/* Copyright 2017 Istio Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef AUTH_CONTROLLER_H
+#define AUTH_CONTROLLER_H
+
+#include "envoy/server/filter_config.h"
+#include "envoy/thread_local/thread_local.h"
+
+#include "config.h"
+#include "http_request.h"
+#include "pubkey_cache.h"
+
+namespace Envoy {
+namespace Http {
+namespace Auth {
+
+// The controller object to handle the token verification flow.
+class Controller : public ThreadLocal::ThreadLocalObject {
```

Comment:
Is this class very specific for JWT, or can be extended to dispatch different 'verify' flow depended on authN type?

Reply:
Yes, it can be extended.